### DISCUSSION TITLE
Fix this??? or stop the warn

### DISCUSSION THREAD

**Discussion by ah-ha-admin** [termux/termux-app#3375] [General]
![8C863E13-5FB6-4950-A287-6F96A3A0FFB2](https://user-images.githubusercontent.com/118419708/236407082-08ea18ae-7086-4ce5-b46c-69a92445c6eb.jpeg)

**Comment by ah-ha-admin** [comment]
@EdwardBetts @bumper314 @jeansch @fornwall

**Reply by jeansch** [reply]
Not a termux issue. RTFM.

**Comment by agnostic-apollo** [comment]
This is not something for us to fix, you have added Microsoft debian repo yourself to apt sources. Linux distro repos are not meant to be used in termux, their packages would (mostly) not be compatible. Remove its entry from the `/data/data/com.termux/files/usr/etc/apt/sources.list` file or files under `/data/data/com.termux/files/usr/etc/apt/sources.list.d`.

The reason for the error is that you have added the key to `trusted.gpg` with `apt-key`, which is deprecated, you instead need to create key under the `/data/data/com.termux/files/usr/etc/apt/trusted.gpg.d` directory and delete it from `trusted.gpg`, plently of resources online for it that can be used as reference. And do not add `sudo` to any commands. And currently its just a warning and not an error and the command should still succeed.

https://losst.pro/en/how-to-fix-key-is-stored-in-legacy-trusted-gpg-keyring-in-ubuntu

https://learn.microsoft.com/en-us/windows-server/administration/linux-package-repository-for-microsoft-software#manual-configuration

https://manpages.ubuntu.com/manpages/impish/en/man8/apt-key.8.html

And do not needlessly tag people in future. You are not owed responses under 24hrs.
